netcore/build.targets

```diff
@@ -0,0 +1,193 @@
+            <Output TaskParameter="ExitCode" PropertyName="_ExitCode" />
+        </Exec>
+    </Target>
+
+    <Target Name="update-roslyn"
+            Condition="!Exists('$(MSBuildThisFileDirectory)roslyn\packages\microsoft.net.compilers.toolset\$(ROSLYN_VERSION)\microsoft.net.compilers.toolset.nuspec')"
+            DependsOnTargets="init-tools">
+        <PropertyGroup>
+                <_UpdateRoslynArgs>&quot;$(MSBuildThisFileDirectory)roslyn-restore.csproj&quot;</_UpdateRoslynArgs>
+                <_UpdateRoslynArgs>$(_UpdateRoslynArgs) -p:RoslynVersion=$(ROSLYN_VERSION)</_UpdateRoslynArgs>
+                <_UpdateRoslynArgs>$(_UpdateRoslynArgs) --packages &quot;$(MSBuildThisFileDirectory)roslyn\packages&quot;</_UpdateRoslynArgs>
+                <_UpdateRoslynArgs>$(_UpdateRoslynArgs) -p:OutputPath=&quot;$(MSBuildThisFileDirectory)roslyn\restore&quot;</_UpdateRoslynArgs>
+            </PropertyGroup>
+        <Exec Command="$(DOTNET) restore $(_UpdateRoslynArgs)">
+            <Output TaskParameter="ExitCode" PropertyName="_ExitCode" />
+        </Exec>
+    </Target>
+
+    <Target Name="update-corefx"
+            Inputs="$(MSBuildThisFileDirectory)corefx-restore.csproj"
+            Outputs="$(MSBuildThisFileDirectory)corefx\.stamp-dl-corefx-$(NETCORETESTS_VERSION)"
+            DependsOnTargets="init-tools">
+        <PropertyGroup>
+            <_UpdateCoreFxArgs>&quot;$(MSBuildThisFileDirectory)corefx-restore.csproj&quot;</_UpdateCoreFxArgs>
+            <_UpdateCoreFxArgs>$(_UpdateCoreFxArgs) --runtime $(RID)</_UpdateCoreFxArgs>
+            <_UpdateCoreFxArgs>$(_UpdateCoreFxArgs) --packages &quot;$(MSBuildThisFileDirectory)corefx\packages&quot;</_UpdateCoreFxArgs>
+            <_UpdateCoreFxArgs>$(_UpdateCoreFxArgs) -p:MicrosoftPrivateCoreFxNETCoreAppVersion=$(NETCORETESTS_VERSION)</_UpdateCoreFxArgs>
+            <_UpdateCoreFxArgs>$(_UpdateCoreFxArgs) -p:OutputPath=&quot;$(MSBuildThisFileDirectory)corefx\restore&quot;</_UpdateCoreFxArgs>
+        </PropertyGroup>
+        <Exec Command="$(DOTNET) build $(_UpdateCoreFxArgs)">
+            <Output TaskParameter="ExitCode" PropertyName="_ExitCode" />
+        </Exec>
+        <Touch Files="$(MSBuildThisFileDirectory)corefx\.stamp-dl-corefx-$(NETCORETESTS_VERSION)" AlwaysCreate="true" />
+    </Target>
+
+    <Target Name="update-netcore-sdk"
+            Condition="!Exists('$(MSBuildThisFileDirectory)$(NETCORESDK_FILE)')">
+        <PropertyGroup>
+            <_UpdateNetCoreSdkArgs>$(MSBuildThisFileDirectory)download-unzip.ps1 -url '$(NETCORE_URL)' -file '$(MSBuildThisFileDirectory)$(NETCORESDK_FILE)' -path '$(MSBuildThisFileDirectory)'</_UpdateNetCoreSdkArgs>
+        </PropertyGroup>
+        <RemoveDir Directories="$(MSBuildThisFileDirectory)shared\Microsoft.NETCore.App" />
+        <Exec Command="powershell -NoProfile -NoLogo -ExecutionPolicy ByPass -Command &quot;$(_UpdateNetCoreSdkArgs)&quot;">
```

Comment:
Why not just use `<Unzip SourceFiles="..." DestinationFolder="..." />` here?

Reply:
We still have VS2015 support for our builds and that task was included in 15.8, so I wanted a solution that would not introduce a dependencies on specific versions (even if we soon will move away from VS2015 up to VS2019), the same goes for download file task that will be needed as well.

Reply:
I am honestly not sure if it's worth the trouble. The CoreFX test downloads already use the task and other .NET Core repositories list VS2019 as requirement for quite some time (VS2017 may still work but VS2015 almost certainly does not).

Even if you wanted to support VS2015 it is probably more reliable to write it as embedded task in C# than relying on shell .zip support.

Reply:
I could convert it into an embedded C# task, dropping the script.

Reply:
VS2015 is not relevant to netcore builds

Reply:
So will it be fine to at least require 15.8 for netcore build? Another option would be to use dotnet msbuild for that specific target and run it similar to how we update corefx tests. Let me do that instead.